### PR TITLE
Support API version v20170131 in apiloader

### PR DESCRIPTION
**Pull request by JiangtianLi**
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/474)

### CODE REVIEW

**Comment by JiangtianLi** [comment]
@acs-bot test this please

File: pkg/api/v20170131/types.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ type ContainerService struct {
 
 // Properties represents the ACS cluster definition
 type Properties struct {
-	ProvisioningState       *ProvisioningState       `json:"provisioningState,omitempty"`
+	ProvisioningState       ProvisioningState        `json:"provisioningState,omitempty"`
```
Comment (rjtsdl) [inline pkg/api/v20170131/types.go]: Reason ?

Comment (rjtsdl) [inline pkg/api/v20170131/types.go]: #456

Comment (JiangtianLi) [inline pkg/api/v20170131/types.go]: In Properties struct, it is also not a pointer. And in other versions, it is not a pointer also. It looks to me a typo here.

File: pkg/api/v20170131/validate.go [hunk 2]
```diff
@@ -84,6 +84,7 @@ func (a *Properties) Validate() error {
 		if agentPoolProfile.OSType == Windows {
 			switch a.OrchestratorProfile.OrchestratorType {
 			case Swarm:
+			case Kubernetes:
```
Comment (JackQuincy) [inline pkg/api/v20170131/validate.go]: /we should do validation that this isn't set in 2016-09-30. Unless it is allowed in the current shipped RP.

Comment (JackQuincy) [inline pkg/api/v20170131/validate.go]: talked off line, I had forgotten this file was only for 2017-01-31. My bad

**Pull request merged**